### ISSUE TITLE
(wrong-type-argument characterp magit-menu-format-desc)

### ISSUE THREAD

**Issue by colonelpanic8** [oantolin/embark#656]
I don't have too much time to investigate this further, but it seems that there is some kind of issue with embark-bindings when running in a magit buffer:

Here is a backtrace:

```
Debugger entered--Lisp error: (wrong-type-argument characterp magit-menu-format-desc)
  embark--command-name((menu-item (magit-menu-format-desc "Apply %x") magit-apply :keys #f(compiled-function (&rest args2) #<bytecode -0xae2890013e3a63f>) :enable (not (memq (magit-diff-type) '(unstaged staged)))))
  embark--formatted-bindings((keymap (keymap (3 keymap (94 keymap ... ... ... ...)) keymap (\1 menu-item (magit-menu-format-desc "Apply %x") magit-apply :keys #f(compiled-function (&rest args2) #<bytecode -0xae2890013e3a63f>) :enable (not ...)) (\2 menu-item (magit-menu-format-desc "Stage %x") magit-stage :keys #f(compiled-function (&rest args2) #<bytecode -0xae2bd176c10a63f>) :enable (eq ... ...)) (\3 menu-item (magit-menu-format-desc "Unstage %x") magit-unstage :keys #f(compiled-function (&rest args2) #<bytecode -0xae2bd249b71a63f>) :enable (eq ... ...)) (\4 menu-item (magit-menu-format-desc "Discard %x") magit-discard :keys #f(compiled-function (&rest args2) #<bytecode -0xae2bd170589a63f>) :enable (not ...)) (\5 menu-item (magit-menu-format-desc "Reverse %x") magit-reverse :keys #f(compiled-function (&rest args2) #<bytecode -0xae2bd2498eea63f>) :enable (not ...)) (\6 menu-item "Visit file" magit-diff-visit-file :keys #f(compiled-function (&rest args2) #<bytecode -0xae2b0bdcd6ea63f>) :enable (memq ... ...)) (\7 menu-item (magit-menu-format-desc "Untrack %x") magit-file-untrack :keys #f(compiled-function (&rest args2) #<bytecode -0xae2b1027439a63f>)) (\8 menu-item "Rename file" magit-file-rename :keys #f(compiled-function (&rest args2) #<bytecode -0xae2b1025740a63f>) :enable (eq ... ...)) (remap keymap (magit-cherry-apply . magit-apply) (magit-stage-file . magit-stage) (magit-unstage-file . magit-unstage) (magit-delete-thing . magit-discard) (magit-revert-no-commit . magit-reverse) (magit-visit-thing . magit-diff-visit-file) (magit-file-untrack . magit-file-untrack) (magit-file-rename . magit-file-rename)) (3 keymap (5 . magit-diff-edit-hunk-commit) (20 . magit-diff-trace-definition)) (67 . magit-commit-add-log) (38 . magit-do-async-shell-command) (24 keymap (97 . magit-add-change-log-entry) (53 keymap ...) (52 keymap ... ...)) (C-return . magit-diff-visit-worktree-file)) (keymap #^[nil nil keymap 
#^^[3 0 nil nil nil (keymap (112 keymap (102 . imalison:projectile-find-file) (115 . imalison:do-rg))) nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil ...] #^^[1 0 #^^[2 0 
#^^[3 0 nil nil nil (keymap ...) nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil ...] nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil ...] nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil] nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil ...]) (keymap (menu-bar keymap (yasnippet menu-item "YASnippet" ... :visible yas-use-menu)) (3 keymap (38 keymap ... ... ...)) (9 menu-item "" yas-expand :filter yas-maybe-expand-abbrev-key-filter) (tab menu-item "" yas-expand :filter yas-maybe-expand-abbrev-key-filter)) (keymap) (keymap) (keymap (11 . magit-gitflow-popup)) (keymap) (keymap (3 keymap (112 . projectile-command-map)) (menu-bar keymap (projectile menu-item "Projectile" ... :visible projectile-show-menu))) (keymap) (keymap) (keymap) (keymap (3 keymap (6 . frame-mode-other-window-or-frame-next-command)) (24 keymap (79 lambda nil ... ...) (111 . frame-mode-other-window) (50 . make-frame) (51 . make-frame))) (keymap (remap keymap (dired-jump . magit-dired-jump)) (106 . magit-status-jump) keymap (menu-bar keymap (magit menu-item "Magit" ...)) (remap keymap (evil-next-line . evil-next-visual-line) (evil-previous-line . evil-previous-visual-line) (next-line . magit-next-line) (previous-line . magit-previous-line)) (23 . magit-copy-section-value) (3 keymap (23 . magit-browse-thing) (15 . magit-browse-thing) (5 . magit-edit-thing) (3 . magit-dispatch)) (62 . magit-sparse-checkout) (33 . magit-run) (36 . magit-process-buffer) (37 . magit-worktree) (90 . magit-worktree) (122 . magit-stash) (89 . magit-cherry) (121 . magit-show-refs) (88 . magit-reset) (120 . magit-reset-quickly) (87 . magit-patch) (119 . magit-am) (86 . magit-revert) (118 . magit-revert-no-commit) (85 . magit-unstage-all) (117 . magit-unstage-file) (84 . magit-notes) (116 . magit-tag) (83 . magit-stage-modified) (115 . magit-stage-file) (82 . magit-file-rename) (114 . magit-rebase) (58 . magit-git-command) (81 . magit-git-command) ...)))
  embark-completing-read-prompter((keymap (keymap (3 keymap (94 keymap ... ... ... ...)) keymap (\1 menu-item (magit-menu-format-desc "Apply %x") magit-apply :keys #f(compiled-function (&rest args2) #<bytecode -0xae2890013e3a63f>) :enable (not ...)) (\2 menu-item (magit-menu-format-desc "Stage %x") magit-stage :keys #f(compiled-function (&rest args2) #<bytecode -0xae2bd176c10a63f>) :enable (eq ... ...)) (\3 menu-item (magit-menu-format-desc "Unstage %x") magit-unstage :keys #f(compiled-function (&rest args2) #<bytecode -0xae2bd249b71a63f>) :enable (eq ... ...)) (\4 menu-item (magit-menu-format-desc "Discard %x") magit-discard :keys #f(compiled-function (&rest args2) #<bytecode -0xae2bd170589a63f>) :enable (not ...)) (\5 menu-item (magit-menu-format-desc "Reverse %x") magit-reverse :keys #f(compiled-function (&rest args2) #<bytecode -0xae2bd2498eea63f>) :enable (not ...)) (\6 menu-item "Visit file" magit-diff-visit-file :keys #f(compiled-function (&rest args2) #<bytecode -0xae2b0bdcd6ea63f>) :enable (memq ... ...)) (\7 menu-item (magit-menu-format-desc "Untrack %x") magit-file-untrack :keys #f(compiled-function (&rest args2) #<bytecode -0xae2b1027439a63f>)) (\8 menu-item "Rename file" magit-file-rename :keys #f(compiled-function (&rest args2) #<bytecode -0xae2b1025740a63f>) :enable (eq ... ...)) (remap keymap (magit-cherry-apply . magit-apply) (magit-stage-file . magit-stage) (magit-unstage-file . magit-unstage) (magit-delete-thing . magit-discard) (magit-revert-no-commit . magit-reverse) (magit-visit-thing . magit-diff-visit-file) (magit-file-untrack . magit-file-untrack) (magit-file-rename . magit-file-rename)) (3 keymap (5 . magit-diff-edit-hunk-commit) (20 . magit-diff-trace-definition)) (67 . magit-commit-add-log) (38 . magit-do-async-shell-command) (24 keymap (97 . magit-add-change-log-entry) (53 keymap ...) (52 keymap ... ...)) (C-return . magit-diff-visit-worktree-file)) (keymap #^[nil nil keymap 
#^^[3 0 nil nil nil (keymap (112 keymap (102 . imalison:projectile-find-file) (115 . imalison:do-rg))) nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil ...] #^^[1 0 #^^[2 0 
#^^[3 0 nil nil nil (keymap ...) nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil ...] nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil ...] nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil] nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil ...]) (keymap (menu-bar keymap (yasnippet menu-item "YASnippet" ... :visible yas-use-menu)) (3 keymap (38 keymap ... ... ...)) (9 menu-item "" yas-expand :filter yas-maybe-expand-abbrev-key-filter) (tab menu-item "" yas-expand :filter yas-maybe-expand-abbrev-key-filter)) (keymap) (keymap) (keymap (11 . magit-gitflow-popup)) (keymap) (keymap (3 keymap (112 . projectile-command-map)) (menu-bar keymap (projectile menu-item "Projectile" ... :visible projectile-show-menu))) (keymap) (keymap) (keymap) (keymap (3 keymap (6 . frame-mode-other-window-or-frame-next-command)) (24 keymap (79 lambda nil ... ...) (111 . frame-mode-other-window) (50 . make-frame) (51 . make-frame))) (keymap (remap keymap (dired-jump . magit-dired-jump)) (106 . magit-status-jump) keymap (menu-bar keymap (magit menu-item "Magit" ...)) (remap keymap (evil-next-line . evil-next-visual-line) (evil-previous-line . evil-previous-visual-line) (next-line . magit-next-line) (previous-line . magit-previous-line)) (23 . magit-copy-section-value) (3 keymap (23 . magit-browse-thing) (15 . magit-browse-thing) (5 . magit-edit-thing) (3 . magit-dispatch)) (62 . magit-sparse-checkout) (33 . magit-run) (36 . magit-process-buffer) (37 . magit-worktree) (90 . magit-worktree) (122 . magit-stash) (89 . magit-cherry) (121 . magit-show-refs) (88 . magit-reset) (120 . magit-reset-quickly) (87 . magit-patch) (119 . magit-am) (86 . magit-revert) (118 . magit-revert-no-commit) (85 . magit-unstage-all) (117 . magit-unstage-file) (84 . magit-notes) (116 . magit-tag) (83 . magit-stage-modified) (115 . magit-stage-file) (82 . magit-file-rename) (114 . magit-rebase) (58 . magit-git-command) (81 . magit-git-command) ...)) nil no-default)
  embark-bindings-in-keymap((keymap (keymap (3 keymap (94 keymap ... ... ... ...)) keymap (\1 menu-item (magit-menu-format-desc "Apply %x") magit-apply :keys #f(compiled-function (&rest args2) #<bytecode -0xae2890013e3a63f>) :enable (not ...)) (\2 menu-item (magit-menu-format-desc "Stage %x") magit-stage :keys #f(compiled-function (&rest args2) #<bytecode -0xae2bd176c10a63f>) :enable (eq ... ...)) (\3 menu-item (magit-menu-format-desc "Unstage %x") magit-unstage :keys #f(compiled-function (&rest args2) #<bytecode -0xae2bd249b71a63f>) :enable (eq ... ...)) (\4 menu-item (magit-menu-format-desc "Discard %x") magit-discard :keys #f(compiled-function (&rest args2) #<bytecode -0xae2bd170589a63f>) :enable (not ...)) (\5 menu-item (magit-menu-format-desc "Reverse %x") magit-reverse :keys #f(compiled-function (&rest args2) #<bytecode -0xae2bd2498eea63f>) :enable (not ...)) (\6 menu-item "Visit file" magit-diff-visit-file :keys #f(compiled-function (&rest args2) #<bytecode -0xae2b0bdcd6ea63f>) :enable (memq ... ...)) (\7 menu-item (magit-menu-format-desc "Untrack %x") magit-file-untrack :keys #f(compiled-function (&rest args2) #<bytecode -0xae2b1027439a63f>)) (\8 menu-item "Rename file" magit-file-rename :keys #f(compiled-function (&rest args2) #<bytecode -0xae2b1025740a63f>) :enable (eq ... ...)) (remap keymap (magit-cherry-apply . magit-apply) (magit-stage-file . magit-stage) (magit-unstage-file . magit-unstage) (magit-delete-thing . magit-discard) (magit-revert-no-commit . magit-reverse) (magit-visit-thing . magit-diff-visit-file) (magit-file-untrack . magit-file-untrack) (magit-file-rename . magit-file-rename)) (3 keymap (5 . magit-diff-edit-hunk-commit) (20 . magit-diff-trace-definition)) (67 . magit-commit-add-log) (38 . magit-do-async-shell-command) (24 keymap (97 . magit-add-change-log-entry) (53 keymap ...) (52 keymap ... ...)) (C-return . magit-diff-visit-worktree-file)) (keymap #^[nil nil keymap 
#^^[3 0 nil nil nil (keymap (112 keymap (102 . imalison:projectile-find-file) (115 . imalison:do-rg))) nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil ...] #^^[1 0 #^^[2 0 
#^^[3 0 nil nil nil (keymap ...) nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil ...] nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil ...] nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil] nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil ...]) (keymap (menu-bar keymap (yasnippet menu-item "YASnippet" ... :visible yas-use-menu)) (3 keymap (38 keymap ... ... ...)) (9 menu-item "" yas-expand :filter yas-maybe-expand-abbrev-key-filter) (tab menu-item "" yas-expand :filter yas-maybe-expand-abbrev-key-filter)) (keymap) (keymap) (keymap (11 . magit-gitflow-popup)) (keymap) (keymap (3 keymap (112 . projectile-command-map)) (menu-bar keymap (projectile menu-item "Projectile" ... :visible projectile-show-menu))) (keymap) (keymap) (keymap) (keymap (3 keymap (6 . frame-mode-other-window-or-frame-next-command)) (24 keymap (79 lambda nil ... ...) (111 . frame-mode-other-window) (50 . make-frame) (51 . make-frame))) (keymap (remap keymap (dired-jump . magit-dired-jump)) (106 . magit-status-jump) keymap (menu-bar keymap (magit menu-item "Magit" ...)) (remap keymap (evil-next-line . evil-next-visual-line) (evil-previous-line . evil-previous-visual-line) (next-line . magit-next-line) (previous-line . magit-previous-line)) (23 . magit-copy-section-value) (3 keymap (23 . magit-browse-thing) (15 . magit-browse-thing) (5 . magit-edit-thing) (3 . magit-dispatch)) (62 . magit-sparse-checkout) (33 . magit-run) (36 . magit-process-buffer) (37 . magit-worktree) (90 . magit-worktree) (122 . magit-stash) (89 . magit-cherry) (121 . magit-show-refs) (88 . magit-reset) (120 . magit-reset-quickly) (87 . magit-patch) (119 . magit-am) (86 . magit-revert) (118 . magit-revert-no-commit) (85 . magit-unstage-all) (117 . magit-unstage-file) (84 . magit-notes) (116 . magit-tag) (83 . magit-stage-modified) (115 . magit-stage-file) (82 . magit-file-rename) (114 . magit-rebase) (58 . magit-git-command) (81 . magit-git-command) ...)))
  embark-bindings(nil)
  funcall-interactively(embark-bindings nil)
  command-execute(embark-bindings)
```

**Comment by oantolin** [comment]
I'll attempt to reproduce and if successful, will investigate.

**Comment by oantolin** [comment]
The error is because magit's keymaps have some extended menu items whose name, rather than a string, is an s-expression. I checked the manual and this is actually allowed: the name is then obtained by evaluating the expression. I added a call to `eval` and the error is gone. 

Unfortunately the menu item's names don't look that great because magit's expressions depend on some dynamic context that magit provides under normal usage, but not when you randomly call embark-bindings. The end result is names like "Apply nil". 🤷🏽
